program/rust/src/tests/test_message.rs

```diff
@@ -1,97 +1,40 @@
 use {
-    crate::accounts::{
-        Message,
-        PriceFeedMessage,
-        TwapMessage,
+    crate::accounts::PythOracleSerialize,
+    byteorder::BigEndian,
+    pythnet_sdk::{
+        messages::{
+            Message,
+            PriceFeedMessage,
+            TwapMessage,
+        },
+        wire::from_slice,
     },
-    quickcheck::Arbitrary,
     quickcheck_macros::quickcheck,
 };
 
-impl Arbitrary for PriceFeedMessage {
-    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
-        let mut id = [0u8; 32];
-        for item in &mut id {
-            *item = u8::arbitrary(g);
-        }
-
-        let publish_time = i64::arbitrary(g);
-
-        PriceFeedMessage {
-            id,
-            price: i64::arbitrary(g),
-            conf: u64::arbitrary(g),
-            exponent: i32::arbitrary(g),
-            publish_time,
-            prev_publish_time: publish_time.saturating_sub(i64::arbitrary(g)),
-            ema_price: i64::arbitrary(g),
-            ema_conf: u64::arbitrary(g),
-        }
-    }
-}
-
 #[quickcheck]
 fn test_price_feed_message_roundtrip(input: PriceFeedMessage) -> bool {
-    let reconstructed = PriceFeedMessage::try_from_bytes(&input.to_bytes());
+    let reconstructed = from_slice::<BigEndian, Message>(&input.to_bytes()).unwrap();
 
     println!("Failed test case:");
     println!("{:?}", input);
     println!("{:?}", reconstructed);
-
-    reconstructed == Ok(input)
-}
-
-
-impl Arbitrary for TwapMessage {
-    fn arbitrary(g: &mut quickcheck::Gen) -> Self {
-        let mut id = [0u8; 32];
-        for item in &mut id {
-            *item = u8::arbitrary(g);
-        }
-
-        let publish_time = i64::arbitrary(g);
-
-        TwapMessage {
-            id,
-            cumulative_price: i128::arbitrary(g),
-            cumulative_conf: u128::arbitrary(g),
-            num_down_slots: u64::arbitrary(g),
-            exponent: i32::arbitrary(g),
-            publish_time,
-            prev_publish_time: publish_time.saturating_sub(i64::arbitrary(g)),
-            publish_slot: u64::arbitrary(g),
-        }
+    match reconstructed {
+        Message::PriceFeedMessage(reconstructed) => reconstructed == input,
+        _ => false,
     }
 }
 
 #[quickcheck]
 fn test_twap_message_roundtrip(input: TwapMessage) -> bool {
-    let reconstructed = TwapMessage::try_from_bytes(&input.to_bytes());
+    let reconstructed = from_slice::<BigEndian, Message>(&input.to_bytes()).unwrap();
```

Comment:
just to be clear about this: If the message struct changes (e.g., we add a field in the SDK), and then we upgrade the dependency here, this test will fail because the generator for the message will be upgraded, but `to_bytes()` will not be. (I think that is a good thing, but I want to make sure that is what will hapeen)

Reply:
Correct